nanoFramework.Azure.Devices.Client/DeviceClient.cs

```diff
@@ -382,25 +387,75 @@ public void RemoveMethodCallback(MethodCallback methodCallback)
         /// Send a message to Azure IoT.
         /// </summary>
         /// <param name="message">The message to send.</param>
+        /// <param name="contentType">Content of the application message.</param>
+        /// <param name="cancellationToken">A cancellation token. If you use the default one, the confirmation of delivery will not be awaited.</param>
+        /// <returns>True for successful message delivery.</returns>
+        public bool SendMessage(
+            string message,
+            string contentType,
+            CancellationToken cancellationToken = default)
+        {
+            return SendMessage(
+                message,
+                contentType,
+                new ArrayList(),
+                cancellationToken,
+                null);
+        }
+
+        /// <summary>
+        /// Send a message to Azure IoT.
+        /// </summary>
+        /// <param name="message">The message to send.</param>
+        /// <param name="contentType">Content of the application message.</param>
+        /// <param name="userProperties">User properties to add to the application message.</param>
         /// <param name="cancellationToken">A cancellation token. If you use the default one, the confirmation of delivery will not be awaited.</param>
         /// <param name="dtdlComponentname">The DTDL component name.</param>
         /// <returns>True for successful message delivery.</returns>
-        public bool SendMessage(string message, CancellationToken cancellationToken = default, string dtdlComponentname = "")
+        public bool SendMessage(
+            string message,
+            string contentType,
+            ArrayList userProperties,
+            CancellationToken cancellationToken = default,
+            string dtdlComponentname = "")
         {
-            string topic = _telemetryTopic;
+            StringBuilder topic = new(_telemetryTopic);
+
+            // add content type to property bag, if there is one
+            if (!string.IsNullOrEmpty(contentType))
+            {
+                topic.Append(EncodeContentType(contentType));
+                topic.Append("&");
+            }
+
+            // add user properties to property bag, if they exist
+            if (userProperties.Count > 0)
```

Comment:
user properties can be null, so would do:
```suggestion
            if (userProperties != null) && (userProperties.Count > 0))
```